Framework/Geometry/src/Instrument/InfoComponentVisitor.cpp

```diff
@@ -81,7 +95,18 @@ void InfoComponentVisitor::registerGenericComponent(
  */
 void InfoComponentVisitor::registerDetector(const IDetector &detector) {
 
-  const size_t detectorIndex = m_assemblySortedDetectorIndices->size();
+  size_t detectorIndex = 0;
+  try {
+    detectorIndex = m_detectorIdToIndexMap->at(detector.getID());
+  } catch (std::out_of_range &) {
+    /*
+     Do not register a detector with an invalid id. if we can't determine
+     the index, we cannot register it in the right place!
+    */
+
+    throw std::runtime_error("Duplicate detector IDs!");
```

Comment:
Not sure I understand this error message. Isn't the check above failing only if the ID is *not* found?